service/surf/apps/materials/utils.py

```diff
@@ -64,21 +64,30 @@ def add_extra_parameters_to_materials(user, materials):
     :param materials: array of materials
     :return: updated array of materials
     """
-    # perhaps this could be optimized by querying the database once for a list of materials
-    # however, we still want to add the 0 values for items we don't have in the database yet so this is easy for now
+
+    material_objects = list(Material.objects.filter(
```

Comment:
```
material_objects = {
    material.external_id: materal
    for material in Material.objects.filter(external_id__in=(m["external_id"] for m in materials)
}
```
Dan kun je in de loop met constante snelheid de lookup doen, i.p.v. elke keer over de lijst met materialen heen.